agent/consul/discoverychain/gateway.go

```diff
@@ -188,17 +195,44 @@ func (l *GatewayChainSynthesizer) Synthesize(chains ...*structs.CompiledDiscover
 // consolidateHTTPRoutes combines all rules into the shortest possible list of routes
 // with one route per hostname containing all rules for that hostname.
 func (l *GatewayChainSynthesizer) consolidateHTTPRoutes() []structs.HTTPRouteConfigEntry {
+	return consolidateHTTPRoutes(l.matchesByHostname, l.suffix, l.gateway)
+}
+
+// FlattenHTTPRoute takes in a route and its parent config entries and returns a list of flattened routes
```

Comment:
Would be ideal if we could explain what "flattened routes" means in this docstring. I feel like I've been in this code a lot and still have to go remind myself what this code is doing every time I see the function name